Scenes/Creeps/Creep.gd

```diff
@@ -497,3 +498,27 @@ func set_spawn_level(spawn_level: int):
 
 func get_special_list() -> Array[int]:
 	return _special_list
+
+
+func set_special_list(special_list: Array[int]):
+	var creep_specials = _creep_specials_container.get_children()
+	creep_specials.clear()
```

Comment:
You are clearing the local array of children. This doesn't actually remove the children from the node, if that's what you are intending to do.

I would suggest to not try to make this function work with multiple calls and instead rename it to "setup_special_list()" and assume that it gets called once per instance.

Reply:
Done